components/fonts/platform/freetype/font_list.rs

```diff
@@ -86,17 +86,29 @@ where
         let mut font_set = FcConfigGetFonts(config, FcSetSystem);
         let font_set_array_ptr = &mut font_set;
         let pattern = FcPatternCreate();
-        assert!(!pattern.is_null());
-        let family_name_cstr: CString = CString::new(family_name).unwrap();
+        if pattern.is_null() {
+            log::error!("Failed to create FcPattern");
+            return;
```

Comment:
I think for cases like in this function, we might want to have a macro that does a debug_assert, and something else (like return here) when debug-assertions are disabled. The reason being that if something is just an error message, it is much more likely to go unnoticed / unfixed, if everything else "still works". In the long term this can lead to many subtle issues.

Reply:
I'd prefer to not add a specific macro for this. I'll add a debug assert instead, like this:

```rust
debug_assert!(!pattern.is_null());
if pattern.is_null() {
    log::error!("Failed to create FcPattern");
    return;
}
```